awx/api/serializers.py

```diff
@@ -2828,25 +2829,27 @@ def to_representation(self, obj):
         return ret
 
     def validate(self, attrs):
-        if 'project' in self.fields and 'playbook' in self.fields:
+        if 'project' in self.fields and 'playbook' in self.fields and 'fqcn_role' in self.fields:
+            fqcn_role = attrs.get('fqcn_role', self.instance.fqcn_role if self.instance is not None else '')
             project = attrs.get('project', self.instance.project if self.instance else None)
             playbook = attrs.get('playbook', self.instance and self.instance.playbook or '')
             scm_branch = attrs.get('scm_branch', self.instance.scm_branch if self.instance else None)
             ask_scm_branch_on_launch = attrs.get('ask_scm_branch_on_launch', self.instance.ask_scm_branch_on_launch if self.instance else None)
-            if not project:
-                raise serializers.ValidationError({'project': _('This field is required.')})
-            playbook_not_found = bool(
-                (project and project.scm_type and (not project.allow_override) and playbook and force_str(playbook) not in project.playbook_files)
-                or (project and not project.scm_type and playbook and force_str(playbook) not in project.playbooks)  # manual
-            )
-            if playbook_not_found:
-                raise serializers.ValidationError({'playbook': _('Playbook not found for project.')})
-            if project and not playbook:
-                raise serializers.ValidationError({'playbook': _('Must select playbook for project.')})
-            if scm_branch and not project.allow_override:
-                raise serializers.ValidationError({'scm_branch': _('Project does not allow overriding branch.')})
-            if ask_scm_branch_on_launch and not project.allow_override:
-                raise serializers.ValidationError({'ask_scm_branch_on_launch': _('Project does not allow overriding branch.')})
+            if fqcn_role in (None, ''):
+                if not project:
+                    raise serializers.ValidationError({'project': _('This field is required.')})
```

Comment:
We don't ... that I'm aware ... have a way to represent something like this in an error "Either do this OR do this other thing"